### PR TITLE
Update metrics use in GpuUnionExec and GpuCoalesceExec

### LINKED ISSUE #2585: [FEA] Improve metrics for GpuUnionExec
**Is your feature request related to a problem? Please describe.**
Related to epic #2334 

`GpuUnionExec` currently records a `totalTime` metric which does not include the cost of executing the child operator but simply captures the cost of incrementing 2 integers for each batch processed. 

**Describe the solution you'd like**
Either record this metric as `opTime` instead of `totalTime` or change the meaning of `totalTime` here to include the cost of executing the child queries.

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by andygrove**
Signed-off-by: Andy Grove <andygrove@nvidia.com>

Closes #2585 and is part of the epic #2334.

GpuUnionExec `totalTime` was just recording the cost of incrementing two counters per batch. I changed this metric to `opTime` and now `totalTime` is the total time, inclusive of executing the input queries.

### CODE REVIEW

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala [hunk 1]
```diff
@@ -408,13 +409,16 @@ case class GpuUnionExec(children: Seq[SparkPlan]) extends SparkPlan with GpuExec
...
     val totalTime = gpuLongMetric(TOTAL_TIME)
 
-    sparkContext.union(children.map(_.executeColumnar())).map { batch =>
-      withResource(new NvtxWithMetrics("Union", NvtxColor.CYAN, totalTime)) { _ =>
-        numOutputBatches += 1
-        numOutputRows += batch.numRows
-        batch
+    withResource(new NvtxWithMetrics("Union", NvtxColor.PURPLE, totalTime)) { _ =>
+      sparkContext.union(children.map(_.executeColumnar())).map { batch =>
+        withResource(new NvtxWithMetrics("Union", NvtxColor.CYAN, opTime)) { _ =>
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala:417]: Does it make sense to have an operation timing for this?  This is literally performing two sums on the CPU, so any value reported will be dominated by metric measurement overhead and not be useful in practice.

Comment (andygrove) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala:417]: It does seem pretty pointless. I will remove it.

**Comment by sameerz** [comment]
@andygrove can this be retargeted to branch-21.08?

**Comment by sameerz** [comment]
Moving to branch-21.08 after @Salonijain27 confirmed with the PR author.

**Review by razajafri — APPROVED** [review]
LGTM

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Pull request merged**